frontend/src/v4/modules/sequences/sequences.redux.ts

```diff
@@ -186,35 +167,31 @@ export const updateFrameWithViewpoint = (state = INITIAL_STATE, { sequenceId, st
 	}
 	Object.assign(frameToUpdate, viewpoint);
 	delete frameToUpdate.state;
-
-	return state;
 };
 
 export const setStepInterval = (state = INITIAL_STATE, { stepInterval }) => {
-	return {...state, stepInterval};
+	state.stepInterval = stepInterval;
 };
 
 export const setStepScale = (state = INITIAL_STATE, { stepScale }) => {
-	return {...state, stepScale};
+	state.stepScale = stepScale;
 };
 
-export const reset = () => {
-	return {...INITIAL_STATE};
+export const reset = (state) => {
+	Object.assign(state, INITIAL_STATE);
```

Comment:
remember that with immer you CAN'T return a state, you just have to modify it

Reply:
@sanmont3drepo The possibility to modify the draft is what makes Immer useful, but you are allowed to return whatever data
https://immerjs.github.io/immer/return
"<i>It is also allowed to return arbitrarily other data from the producer function. But only if you didn't modify the draft. This can be useful to produce an entirely new state</i>"

Reply:
When we mixed between returning and modifying a state. Immer complained that you had to picked one.

Reply:
In tested it and I stand corrected.

When he have stuff like this in the **same** function is when it fails
<img width="455" alt="image" src="https://github.com/3drepo/3drepo.io/assets/42032433/ab6daa62-7a6f-4798-b207-c8158cc9ae3a">

<img width="765" alt="image" src="https://github.com/3drepo/3drepo.io/assets/42032433/a5229216-f828-4f1a-8647-702640830713">